components/camel-wal/src/main/java/org/apache/camel/component/wal/LogWriter.java

```diff
@@ -239,7 +240,7 @@ public void updateState(EntryInfo.CachedEntryInfo entryInfo, LogEntry.EntryState
 
          Trying to update a persisted entry here is not acceptable
          */
-        assert layerInfo != null;
+        ObjectHelper.notNull(layerInfo, "layerInfo");
```

Comment:
This is also not correct and also impacts the performance of the log writer as the assert is native and would not be enabled in production code (whereas the ObjectHelper will incur the cost of `notNull` check on a critical section of this code).

Reply:
> impacts the performance of the log write

Not sure if a null check through `ObjectHelper.notNull`will have any reasonable performance impact, or do we maybe have any benchmark for that which proves it otherwise?

Reply:
The impact of `ObjectHelper.nonNull` is pretty small. Almost close to none, but still a bit faster. 

```
AssertTest.benchmarkAssert   avgt   10  183.115 ± 0.844  us/op
AssertTest.benchmarkHelper   avgt   10  186.937 ± 0.779  us/op
```

Of all the items I commented, this one is the only one which is a big -1 from me. The log writer adds a small bit of latency on the Resume process due to the I/O, so every bit of performance that can be saved here, is important.